aten/src/ATen/Utils.h

```diff
@@ -26,17 +26,16 @@ namespace at {
 
 AT_API int _crash_if_asan(int);
 
-template <typename T>
-static inline T* checked_cast_storage(Storage* expr, const char * name, int pos, Backend backend, ScalarType scalar_type) {
-  if (at::detail::get_backend(expr->pImpl()) != backend) {
-    AT_ERROR("Expected object of backend ", backend, " but got backend ", at::detail::get_backend(expr->pImpl()),
+static inline const Storage& checked_storage(const Storage& expr, const char * name, int pos, Backend backend, ScalarType scalar_type) {
+  if (at::detail::get_backend(expr.get()) != backend) {
```

Comment:
You don't need to fix this now -- but this should take in a device type, no?

Reply:
yep